src/udiskslinuxmanager.c

```diff
@@ -848,6 +852,8 @@ handle_mdraid_create (UDisksManager         *_object,
   return TRUE; /* returning TRUE means that we handled the method invocation */
 }
 
+#endif /* HAVE_MDRAID */
```

Comment:
This is a really long *ifndef-endif* block. It would be extremely hard to notice it in the codebase when doing future changes. I think we need to put these functions elsewhere and make just including them guarded with *ifndef*.

Reply:
Agreed, the long ifdefs are nasty. One other possible way would be to implement empty symbols (e.g. for `bd_md_` functions) just to satisfy compiler needs - in the light of my other suggestion to return proper error at the beginning of this D-Bus method call handler, this would be a dead code anyway. The result is clean with no ifdefs other than the empty compat symbol declarations.